community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/SchemaLayout.java

```diff
@@ -89,8 +89,9 @@ public final int compare( KEY o1, KEY o2 )
         int valueComparison = compareValue( o1, o2 );
         if ( valueComparison == 0 )
         {
-            // This is a special case where we need also compare entityId to support inclusive/exclusive
-            if ( o1.getCompareId() & o2.getCompareId() )
+            // It's the first key that decides whether or not to compare entityIds, i.e. the level of uniqueness.
+            // The first key is the key that is "the" key that is being worked with and the other key is an existing key being compared with.
+            if ( o1.getCompareId() )
```

Comment:
I don't agree with this. Compare method should yield same logical result independent of order of parameters. If `A > B` then `B < A`.

Reply:
You're absolutely right, this isn't right